### PR TITLE
Address a11y violations in Resources screen(s)

### LINKED ISSUE #966: [A11y] Address A11y issues on Resources
This ticket will look to address issues of an accessibility (a11y) nature in the Resources section of the main nav.

> Get the [Chrome Dev Tools Axe Extension](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd)
> Running this extension on the page first loaded and as you use the page will help reveal what needs to be fixed

## Areas covered by this ticket

Main nav: Resources

## Data setup for testing

There is no additional data setup needed for this page.

### PR DESCRIPTION
**Pull request by jenny-s51**
## Description

Closes #966 

## How Has This Been Tested?

Before (Card View):
Screen Shot 2023-02-21 at 3 56 35 PM

After (Card View): 
Screen Shot 2023-02-21 at 3 42 44 PM

Before (List View):
Screen Shot 2023-02-21 at 3 57 32 PM

After (List View):
Screen Shot 2023-02-21 at 3 42 55 PM

Before and After Empty State - no violations:
Screen Shot 2023-02-21 at 3 54 03 PM

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/components/OdhDocListItem.tsx
Comment (Gkrumbach07) [inline frontend/src/components/OdhDocListItem.tsx:96]: should this be removed or just changed to be unique?

Comment (jenny-s51) [reply frontend/src/components/OdhDocListItem.tsx:96]: Updated to use a unique ID here as discussed in scrum today @Gkrumbach07!

File: frontend/src/components/OdhDocCard.tsx
Comment (lucferbux) [inline frontend/src/components/OdhDocCard.tsx:138]: I presume we are removing this prop since it's deprecated right?
Dont know why we kept it since we already had isSelectable.

**Review by lucferbux — APPROVED** [review]
Great Jenny thank you very much for the first PR, no questions so far everything seems very straight forward.

File: frontend/src/pages/learningCenter/LearningCenterDataView.tsx
Comment (DaoDaoNoCode) [inline frontend/src/pages/learningCenter/LearningCenterDataView.tsx:104]: I think it doesn't hurt to use `doc.spec.appDisplayName` here though I prefer `doc.metadata.name` or `doc.metadata.uid`. The code LGTM! Thanks @jenny-s51

**Review by DaoDaoNoCode — APPROVED** [review]

File: frontend/src/components/OdhDocCard.tsx
Comment (jenny-s51) [reply frontend/src/components/OdhDocCard.tsx:138]: Yes that's correct @lucferbux !

File: frontend/src/components/OdhDocListItem.tsx
Comment (andrewballantyne) [inline frontend/src/components/OdhDocListItem.tsx:102]: Why did we do this change? Was there an issue with A11y?

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/src/pages/learningCenter/LearningCenterDataView.tsx
Comment (andrewballantyne) [reply frontend/src/pages/learningCenter/LearningCenterDataView.tsx:104]: Yeah -- okay, took forever to fix my server and get the test in a place I could verify this.

Error from React:
```
Encountered two children with the same key, `jupyter-doc`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```

I can cause this error due to the way the information is processed. We'll need to get the `${type}-${name}` going on for key as well. 

Side note: It does render both cards, just with the error.
image

File: frontend/src/components/OdhDocListItem.tsx
Comment (jenny-s51) [reply frontend/src/components/OdhDocListItem.tsx]: As discussed, all ids in this screen should be unique (per aXe devtools)!

File: frontend/src/pages/learningCenter/LearningCenterDataView.tsx
Comment (jenny-s51) [reply frontend/src/pages/learningCenter/LearningCenterDataView.tsx]: Nice catch @andrewballantyne , updated based on your feedback! Let me know if these changes [here](https://github.com/opendatahub-io/odh-dashboard/pull/970/commits/37069823facde07bc51d1b4709a177cebe21810f) are what you had in mind.

**Review by andrewballantyne — APPROVED** [review]
/hold

Looks good, I'd like to test it in the UI first before merging. Might be Monday before I do that.

**Comment by andrewballantyne** [comment]
Tested, looks good.

/unhold

**Pull request merged**
